### ISSUE TITLE
set-alias functionality (sh functions)

### ISSUE THREAD

**Issue by nanobowers** [cea-hpc/modules#193]
Referencing issues #107 and #127..

I'm trying to convert my modules 3.2.x environment to v4.x and am running into issues with reduced functionality of set-alias.

As documented, it no longer will create sh functions, instead only creating a sh alias.  I am sure it was difficult to support output of alias and function consistently.

My use case is that I have a function that I believe requires evaluating another script, so that the function can be used to conditionally change-directory within the active shell.

For example, my shell function is:
```sh
function mycd () { eval `mycd.sh $1` ; }
```
And the contrived script is:
```sh
if [ -z "$HOME" ]; then
    echo "echo \"home variable must be set\""
elif [ ! -z "$1" ]; then
    newdir="$HOME/$1"
    if [ -e $newdir ]; then
	echo "cd $newdir"
    else
	echo "echo 'cannot find dir $1 in home space'"
    fi
else
    echo "echo \"must provide an argument\""
fi
```
Sometimes the `eval` will be used to echo a message, other times it will evaluate `cd some_directory`
More complicated versions of this may use a python/ruby script, to generate the thing to `eval`.

I think this is how **modules** works with sh shells (defining a function that does evals).

I cannot think of a good workaround with modules 4.x.  If there is another way to do this, can you let me know the intended workaround?  If not, would you be willing to consider pull requests that add sh-function ability to modules 4.x, either with set-alias, or perhaps a new keyword?

**Comment by xdelaruelle** [comment]
`set-alias` is now a command that only produce shell alias as the test used in Modules 3.x to know whether defining a shell alias or a shell function was not consistent (in some cases people wished to get a shell alias but a shell function was produced).

So as you spot, there is currently no modulefile command to produce shell function. So I will totally consider pull requests for that, preferably with new keywords named `set-function` and `unset-function`.

**Comment by nanobowers** [comment]
created PR #194

**Comment by xdelaruelle** [comment]
Fixed by PR #194
